NachoSpigot-Server/src/main/java/net/minecraft/server/PlayerConnection.java

```diff
@@ -2074,7 +2074,7 @@ public void a(PacketPlayInUpdateSign packetplayinupdatesign) {
 
             TileEntitySign tileentitysign = (TileEntitySign) tileentity;
 
-            if (!tileentitysign.b() || tileentitysign.c() != this.player) {
+            if (!tileentitysign.b() || tileentitysign.signEditor == null || !tileentitysign.signEditor.equals(this.player.getUniqueID())) { // Paper
```

Comment:
isn't this just what .c() != this.player did?

Reply:
`.c() != this.player` compares the players, while this compares their UUID